### ISSUE TITLE
compareDateRange error while updating timeDimensions

### ISSUE THREAD

**Issue by SoozaV** [cube-js/cube#6954]
I'm trying to create a compareDateRange chart using QueryBuilder, I'm using RangePicker (antd) to select a date range and then pass that value in the update method:

```
<RangePicker
   onChange={(e) => {
      if (e?.[0] && e?.[1]) {
          let dateRange = [e[0].format("YYYY-MM-DD"), e[1].format("YYYY-MM-DD")] as DateRange;

         //@ts-ignore without this it says -> Property 'compareDateRange' does not exist on type 'TimeDimension'. Property 'compareDateRange' does not exist on type 'TimeDimensionRanged'.
         let compareDateRange: DateRange[] = query.timeDimensions?.[0].compareDateRange || [];
         compareDateRange.push(dateRange);
         console.log(compareDateRange)
         
         timeDimensions.map((td) => {
               updateTimeDimensions.update(td, {
                   ...td,
                   compareDateRange                                          
               })
         })
      }
   }}
 />
```
When I console.log the compareDateRange var it looks right:

![image](https://github.com/cube-js/cube/assets/23610892/beb383cc-0092-43cf-9a9a-1559abb01c73)

but the output query is:

```
{
   "measures": "...",
   "timeDimensions": [
      {
         "dimension":"...",
         "granularity":"...",
         "compareDateRange": [
            { "0":"2022-03-01","1":"2022-03-31"} <- ???
         ]
      }
   ],
}
```
Why is returning an Array of objects?

Also I get this query Error:
`Error: Invalid query format: child "timeDimensions" fails because ["timeDimensions" at position 0 fails because [child "dateRange" fails because ["dateRange" must be an array, "dateRange" must be a string]]]`

Version: 0.33.12

**Comment by SoozaV** [comment]
@vasilev-alex a little help would be appreciated 😢

**Comment by vasilev-alex** [comment]
Closed by #8339
